### ISSUE TITLE
bug(autocomplete): Firefox blocks popup opened by Enter (onOptionSelected -> window.open)

### ISSUE THREAD

**Issue by enmy726** [angular/components#23727]
#### Reproduction

Steps to reproduce:
1. Open autocomplete and select item by key arrows
2. Click Enter

Result: onOptionSelected was called, popup was blocked.

Example code onOptionSelected function:
onOptionSelected(event: MatAutocompleteSelectedEvent): void {
  window.open('http://localhost', '_blank');
}

#### Expected Behavior
FireFox doesn't block window.open on Enter click. 

#### Actual Behavior
It works on mouse click, but blocked on Enter. 
It works in Chrome on Enter.
I don't have any async functions inside onOptionSelected function.

#### Environment
- Angular: 
- CDK/Material: "12.1.1",
- Browser(s): Firefox
- Operating System (e.g. Windows, Ubuntu):

**Comment by jelbourn** [comment]
I don't believe this is caused by anything in Angular Material versus just being the behavior of Firefox. Feel free to open another issue if you can point to something that indicates it's a problem in the library.
